### PR TITLE
feat: trackCard component added and audio tests added

### PR DESCRIPTION
**Pull request by Kush-Wednesday**
TrackCard Component rendered inside TrackList Component and audio unit tests written.

image

image

image

### CODE REVIEW

File: features/itunes/components/TrackCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          id="track_name"
+          values={{ name: !isUndefined(result.trackName) && result.trackName.substring(0, 10) }}
+        />
+      </TrackName>
+      <ArtistName>
+        <BlockText id="artist_name" values={{ name: result.artistName.substring(0, 10) }} />
+      </ArtistName>
+      <Duration>
+        <BlockText id="duration" />
+        {Math.floor(result.trackTimeMillis / 60000)}:
```
Comment (christin-wednesday) [inline features/itunes/components/TrackCard/index.tsx]: Please add null checks.

File: features/itunes/components/TrackCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    <CustomTrackCard data-testid="track-card">
+      <AlbumArt data-testid="album-art" src={result.artworkUrl100}></AlbumArt>
+      <TrackName>
+        <BlockText
+          id="track_name"
+          values={{ name: !isUndefined(result.trackName) && result.trackName.substring(0, 10) }}
+        />
+      </TrackName>
+      <ArtistName>
+        <BlockText id="artist_name" values={{ name: result.artistName.substring(0, 10) }} />
```
Comment (christin-wednesday) [inline features/itunes/components/TrackCard/index.tsx]: Why is values an object ?
Can we update Blocktext to take in a prop instead ?

```javascript

<BlockText name="something-here" />


```

File: features/itunes/components/TrackCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+} from "@features/itunes/components/styled";
+import { isUndefined } from "lodash";
+import { injectIntl, IntlShape } from "react-intl";
+import { T } from "../../../../common";
+
+interface TrackCardProps {
+  intl: IntlShape;
+  result: SongItem;
+  memoizedAudioRef;
+  trackEventHandler: (x: any) => void;
```
Comment (christin-wednesday) [inline features/itunes/components/TrackCard/index.tsx]: please avoid any definitions

File: features/itunes/components/TrackCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      </Duration>
+      <audio
+        id={result.previewUrl}
+        data-testid="audio-test"
+        autoPlay={false}
+        controls
+        onPlay={() => trackEventHandler(result.trackId)}
+        ref={memoizedAudioRef ? memoizedAudioRef?.[result.trackId] : null}
+        style={
+          {
```
Comment (christin-wednesday) [inline features/itunes/components/TrackCard/index.tsx:54]: Please use styled-components to customize the audio element.

File: features/itunes/components/TrackList/index.tsx [hunk 5]
```diff
@@ -74,40 +60,11 @@ const TrackList: React.FC<TrackListProps> = props => {
...
-                        width: "fit-content<100%>",
-                        maxWidth: "100%",
-                      } as React.CSSProperties
-                    }
-                  >
-                    <source src={item.previewUrl} type="audio/mp3"></source>
-                    Your browser does not support audio tags
-                  </audio>
-                </CustomTrackCard>
+                <TrackCard
```
Comment (christin-wednesday) [inline features/itunes/components/TrackList/index.tsx:62]: Do we have a shimmer while the tracks are loading ?

**Review by christin-wednesday — CHANGES_REQUESTED** [review]
What happens if the tracks are not loaded ?

File: features/itunes/components/TrackCard/tests/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  });
+
+  it("should match the snapshot", () => {
+    const { baseElement } = render(
+      <TrackCard {...trackCardProps} trackEventHandler={trackEventSpy} />
+    );
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it(" display dom for debug", () => {
```
Comment (christin-wednesday) [inline features/itunes/components/TrackCard/tests/index.test.tsx]: Please add assertions here. What do we expect if the dom is created ?

File: features/itunes/components/TrackCard/tests/index.test.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    const { debug } = render(<TrackCard {...trackCardProps} trackEventHandler={trackEventSpy} />);
+    debug();
+  });
+
+  it("should check if image is rendered", () => {
+    const { getAllByTestId } = render(
+      <TrackCard {...trackCardProps} trackEventHandler={trackEventSpy} />
+    );
+    const albumArt = getAllByTestId("album-art")[0] as HTMLImageElement;
+    expect(albumArt.src).toContain(
```
Comment (christin-wednesday) [inline features/itunes/components/TrackCard/tests/index.test.tsx:40]: nice 🚀

File: features/itunes/components/TrackCard/tests/index.test.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+      "https://is5-ssl.mzstatic.com/image/thumb/Music116/v4/5a/47/6d/5a476ddd-4690-1297-1896-6a286a497a21/191404113974.png/100x100bb.jpg"
+    );
+  });
+
+  it("should render audio element", async () => {
+    const { getAllByTestId, debug } = await render(
+      <TrackCard {...trackCardProps} trackEventHandler={trackEventSpy} />
+    );
+    const audioElement = getAllByTestId("audio-test")[0] as HTMLAudioElement;
+    debug(audioElement);
```
Comment (christin-wednesday) [inline features/itunes/components/TrackCard/tests/index.test.tsx]: We should remove debug invocations

File: features/itunes/components/TrackCard/tests/index.test.tsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    fireEvent.play(audioElement);
+    expect(trackEventSpy).toBeCalled();
+    expect(audioElement.id).toContain("22233223");
+    expect(audioElement.controls).toBeTruthy();
+    fireEvent.pause(audioElement);
+    fireEvent.play(audioElement);
+    expect(trackEventSpy).toBeCalledTimes(2);
+  });
+});
```
Comment (christin-wednesday) [inline features/itunes/components/TrackCard/tests/index.test.tsx:80]: Please add more tests

**Review by christin-wednesday — CHANGES_REQUESTED** [review]

File: features/itunes/components/TrackList/index.tsx (hunk 5 above)
Comment (Kush-Wednesday) [reply features/itunes/components/TrackList/index.tsx:62]: like skeleton tag from antd?

**Pull request merged**